docs/design/podService.md

```diff
@@ -0,0 +1,94 @@
+# Services for Addons Pods
+
+The Kubernetes CSI Addons project currently connects the CSI Addons Manager to the add-on sidecar through hardcoded IP addresses. However, this IP-based approach poses limitations, especially when implementing secure TLS communication between the manager and the sidecar. TLS certificates typically require stable DNS-based connections rather than IP-based, as IP addresses can change and are not easily resolved in cases where host networking is enabled.
+
+## Problem Statement
+
+1. **IP Dependency**: The current approach hardcodes IP addresses, which can be unreliable and lead to broken connections if IPs change.
+2. **TLS Challenges**: Introducing TLS certificates requires a stable DNS name for validation. Relying on IPs does not align with best practices in certificate management and TLS.
+3. **Host Networking Constraints**: Host networking limits the use of Pod DNS names, as Pod DNS entries may not resolve correctly or may be inaccessible.
+
+## Proposed Solution
+
+### 1. Exposing Add-ons Sidecar via Kubernetes Service
+
+To address the above challenges, this proposal suggests creating a dedicated Kubernetes Service for each add-ons sidecar Pod. This Service will:
+
+- Provide a consistent DNS name, eliminating the dependency on IP addresses.
+- Enable TLS certificates to be issued with a stable DNS name.
+- Allow secure connectivity between the manager and the sidecar by abstracting away the networking specifics, especially when host networking is enabled.
+
+The Service could be of type `ClusterIP`, as the primary objective is intra-cluster communication between the manager and sidecar.
```

Comment:
If we are trying to stick to a single IP for accessing the pod, using clusterIP can be a problem for edge case where the service is deleted and recreated.